### PR TITLE
Added ability to compile lua as C++

### PR DESCRIPTION
**Pull request by dedmen**
Mainly to be able to make LUA use C++ exceptions instead of longjmp.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.1)
 project(lua LANGUAGES C VERSION 5.4.2)
 
 option(LUA_SUPPORT_DL "Support dynamic loading of compiled modules" OFF)
+option(LUA_BUILD_AS_CXX "Build lua as C++" OFF)
```
Comment (walterschell) [inline CMakeLists.txt:5]:
```suggestion
option(LUA_BUILD_AS_CXX "Build lua as C++" OFF)
enable_language(CXX)
```

**Comment by walterschell** [comment]
Thank you for the PR. I've made a new branch with updated GitHub actions to land this. Would you mind applying the suggested change, and then rebasing your PR so that it picks up the new pipeline?

**Comment by walterschell** [comment]
Looks like the compiler and bytecode compiler are the culprits. If you default them off if compiling as cxx at top level, then FATAL_ERROR them when built in case they were manually set in cache, I think that would take care of it

**Comment by dedmen** [comment]
Well I only tested lua static build as thats what I was using :upside_down_face: and that seems to work

**Comment by dedmen** [comment]
compiling luac.c and lua.c as c++ is probably better than excluding them

**Comment by walterschell** [comment]
I wasn't sure that `lua.c` and `luac.c` also built as C++. Evidently they do. Thanks again for doing the MR.

**Pull request merged**